digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -613,22 +613,10 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
     Kalpa-API methods
   */
   def getRoadAddressesLinkByMunicipality(municipality: Int, roadLinkDataTempAPI:Boolean=false): Seq[RoadAddressLink] = {
-    //TODO: Remove null checks and make sure no nulls are generated
-    //TODO: All this request to VVH can be done using asynchronous
-    val roadLinks = {
-      val tempRoadLinks = roadLinkService.getViiteRoadLinksFromVVHByMunicipality(municipality,frozenTimeVVHAPIServiceEnabled)
-      if (tempRoadLinks == null)
-        Seq.empty[RoadLink]
-      else tempRoadLinks
-    }
-    val complimentaryLinks = {
-      val tempComplimentary = roadLinkService.getComplementaryRoadLinksFromVVH(municipality)
-      if (tempComplimentary == null)
-        Seq.empty[RoadLink]
-      else tempComplimentary
-    }
+
+    val (roadLinksWithComplementary, _) = roadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(municipality)
```

Comment:
are we loosing frozentime boolean effect here though?

Reply:
yes, i will put a clause to use the frozen method if flag is active, but it will be without the cache.